src/backend/gl/src/device.rs

```diff
@@ -1011,47 +1053,15 @@ impl d::Device<B> for Device {
         offset: u64,
         buffer: &mut n::Buffer,
     ) -> Result<(), d::BindError> {
-        let gl = &self.share.context;
-        let target = buffer.target;
-
-        if offset == 0 {
-            memory.first_bound_buffer.set(Some(buffer.raw));
-        } else {
-            assert!(memory.first_bound_buffer.get().is_some());
-        }
-
-        let cpu_can_read = memory.can_download();
-        let cpu_can_write = memory.can_upload();
-
-        if self.share.private_caps.buffer_storage {
-            //TODO: glow::DYNAMIC_STORAGE_BIT | glow::MAP_PERSISTENT_BIT
-            let flags = memory.map_flags();
-            //TODO: use *Named calls to avoid binding
-            gl.bind_buffer(target, Some(buffer.raw));
-            gl.buffer_storage(target, buffer.requirements.size as _, None, flags);
-            gl.bind_buffer(target, None);
-        } else {
-            let flags = if cpu_can_read && cpu_can_write {
```

Comment:
Removing these flags appears to be a regression. It's fine to catch up to this after the fact, since your change is much more important, but need to keep it on the radar.

Reply:
The way the gl backend was before, allocated memory had `n::Memory` properties hard-coded to `memory::Properties::CPU_VISIBLE | memory::Properties::CPU_CACHED`, which means that it was always really allocated with `glow::MAP_READ_BIT | glow::MAP_WRITE_BIT` or `glow::DYNAMIC_DRAW`, the rest of it was dead code.

Now, it's still hard-coded, but since the allocation takes place in the same place that the `n::Memory` flags are set, there's nothing to check to to see what flags to use when allocating.

I agree it would probably be better if you could control the buffer flags more than this, but it should have the same behavior as it did before.